### DISCUSSION TITLE
how to apply boundary condition on an auxiliary variable?

### DISCUSSION THREAD

**Discussion by zuckarm** [idaholab/moose#27657] [Q&A Modules: Phase field] [answered]
hi all, 
i am solving a phase field problem, the geometry is that of an edge-cracked body. i have defined a sideset 'crack', and i want to apply the condition that phase field auxiliary variable d = 1 on 'crack'. can anyone please guide on how to execute this?

**Comment by GiudGiud** [comment]
Hello

You can apply boundary conditions to auxiliary variables.
This seems like a DirichletBC. 

Guillaume

**Reply by zuckarm** [reply]
hi, sorry for the late response.
i have applied DirichletBC, as you have suggested.
code snippet:
    [crack]
        type = DirichletBC
        variable = d
        boundary = crack
        value = 1
    []
i am getting the following error: No solver variable named d found. Did you specify an auxiliary variable when you meant to specify a solver variable?

**Reply by maxnezdyur** [reply, marked as answer]
I think you just want to run an AuxKernel that sets the changes the value of the AuxVariable on the boundary. AuxKernels can be boundary restricted too. So you just use the `ConstantAux` and the `boundary = crack`. The only problem is that if you have multiple auxkernels for the same variable they can override each other.
